server/util/mapKeysDeep.js

```diff
@@ -0,0 +1,25 @@
+import R from 'ramda';
+
+/**
+ * Recursively traverses the object and applies a function to all the keys.
+ * This function is curried.
+ * @param {* Function} func
+ * @param {* Object} value
+ */
+function mapKeysDeep(func, value) {
```

Comment:
When recursively traversing objects keys and values in JavaScript, bear in mind that the value can be cyclic. Does your `mapKeysDeep` recurse infinitely when there are circular references?

Lodash `_.flattenDeep` does not handle circular references but `_.cloneDeep` does. It'll be up to you whether you want to support it. If you are asked to write similar traversal code in interviews, this is a good point to point out to impress the interviewer!

If you decide not to support it, leave a warning in the source code (: